ts/haskell/mod.ts

```diff
@@ -37,23 +37,38 @@ type Execs<Exe extends string[]> = {
  * @param description - A short description of the project
  * @param compiler - The compiler version (e.g.: "ghc94")
  * @param executables - The name of the executables in the cabal file
+ * @param overrideDependencies - A mapping from package names to versions
  * @param src - The source directory
  */
 export function mkHaskellProject<const Executables extends string[]>(args: {
   description: string;
   compiler: string;
   executables?: Executables;
+  overrideDependencies?: Record<string, string>;
   src: string;
 }): Project & HaskellAddenda & Execs<Executables> {
-  const pkg: Package = mkPackage(
-    nixRaw`
-      (pkgs.haskell.packages.${nixRaw(args.compiler)}.callCabal2nix
+  const deps = args.overrideDependencies ? args.overrideDependencies : {};
+
+  const nixPkg = nixRaw`
+    let
+      haskell = pkgs.haskell.packages.${nixRaw(args.compiler)}.override {
+        overrides = final: prev:
+          ${nixAttrSet(
+            mapValues(
```

Comment:
Pushed 2dfd9f7c, which I think is better. TIOLI.

Reply:
Much better!